do/droplets.go

```diff
@@ -174,3 +184,24 @@ func (i *instances) dropletByName(ctx context.Context, nodeName types.NodeName)
 
 	return nil, cloudprovider.InstanceNotFound
 }
+
+// dropletIDFromProviderID returns a droplet's ID extracted from the node's
+// providerID spec. The providerID spec should be retrievable from the Kubernetes
+// node object. The expected format is: digitalocean://droplet-id
+func dropletIDFromProviderID(providerID string) (string, error) {
+	if providerID == "" {
+		return "", errors.New("providerID cannot be empty string")
+	}
+
+	split := strings.Split(providerID, "/")
+	if len(split) != 3 {
+		return "", fmt.Errorf("unexpected providerID format: %s, format should be: digitalocean://12345", providerID)
+	}
+
+	// since split[0] is actually "digitalocean:"
+	if strings.TrimSuffix(split[0], ":") != providerName {
+		return "", fmt.Errorf("provider name from providerID should be digitalocean: %s", providerID)
+	}
```

Comment:
`return "", fmt.Errorf("provider name from providerID should be %s://%s", providerName, providerID)`?

Reply:
`providerID` is already in the format `providerName://id` so it would be `digitalocean://digitalocean://droplet-id`. I think the confusion here is that `providerID` is the droplet ID which it is not, it's a separate concept from Kubernetes land :P